src/core/main/worker/worker_main.ts

```diff
@@ -307,6 +308,13 @@ export default function initializeWorkerMain() {
         break;
       }
 
+      case MainThreadMessageType.CdmCodecSupportUpdate: {
+        for (const { mimeType, codec, result } of msg.value) {
+          cdmCodecSupportProber.addToCache(mimeType, codec, result);
```

Comment:
Why not just go through `CodecSupportUpdate` with a supplementary information, as it's already its role?

Also the issue I see with a `cdmCodecSupportProber` is that it might depend on the current key system, no?